contracts/AelinFeeEscrow.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract AelinFeeEscrow {
+    using SafeERC20 for IERC20;
+    uint256 public vestingExpiry;
+    address public treasury;
+    address public futureTreasury;
+    address public escrowedToken;
+
+    bool private calledInitialize;
+
+    /**
+     * @dev the constructor will always be blank due to the MinimalProxyFactory pattern
+     * this allows the underlying logic of this contract to only be deployed once
+     * and each new escrow created is simply a storage wrapper
+     */
+    constructor() {}
+
+    /**
+     * @dev the treasury may change their address
+     */
+    function setTreasury(address _treasury) external onlyTreasury {
+        futureTreasury = _treasury;
+    }
+
+    function acceptTreasury() external {
+        require(msg.sender == futureTreasury, "only future treasury can access");
+        treasury = futureTreasury;
+        emit SetTreasury(futureTreasury);
+    }
+
+    function initialize(address _treasury, address _escrowedToken) external initOnce {
```

Comment:
we should emit an event on intialization too

Reply:
see 4f84e871ad166bfdfc65e068ffa1b54f027c06bb